libbpfgo.go

```diff
@@ -1302,6 +1304,31 @@ func (p *BPFProg) SetAttachType(attachType BPFAttachType) {
 	C.bpf_program__set_expected_attach_type(p.prog, C.enum_bpf_attach_type(int(attachType)))
 }
 
+func (p *BPFProg) AttachCgroup(cgroupV2DirPath string) (*BPFLink, error) {
+	const (
+		O_DIRECTORY int = 0200000
+		O_RDONLY    int = 00
+	)
+	fd, err := syscall.Open(cgroupV2DirPath, O_DIRECTORY|O_RDONLY, 0)
+	if fd < 0 {
+		return nil, fmt.Errorf("failed to open cgroupv2 directory path %s: %w", cgroupV2DirPath, err)
+	}
+	link := C.bpf_program__attach_cgroup(p.prog, C.int(fd))
+	if C.IS_ERR_OR_NULL(unsafe.Pointer(link)) {
+		return nil, errptrError(unsafe.Pointer(link), "failed to attach cgroup on cgroupv2 %s to program %s", cgroupV2DirPath, p.name)
+	}
+
+	dirName := strings.ReplaceAll(cgroupV2DirPath[1:], "/", "-")
```

Comment:
can you explain (or event better - add a comment) - why do we need that replace?

Reply:
dirName will be used to create bpfLink.eventName, which is used for the link pinning. We can't have the link name with / because it would be confused with the filesystems /s. I'll add a comment to clarify.